tests/execution/test-http-batchKey.md

```diff
@@ -0,0 +1,90 @@
+# Http with args as body
+
+```graphql @config
+schema
+  @server(port: 8000)
+  @upstream(baseURL: "http://jsonplaceholder.typicode.com", batch: {maxSize: 1000, delay: 10}) {
+  query: Query
+}
+
+type Query {
+  foos: FooResponse @http(path: "/foo")
+}
+
+type FooResponse {
+  foos: [Foo!]!
+}
+
+type Foo {
+  id: ID!
+  fooName: String!
+  barId: String!
+  bars: [Bar!]!
+    @http(
+      path: "/bar"
+      query: [{key: "barId[]", value: "{{.value.barId}}"}]
+      batchKey: "barId[]"
+      batchKey: ["bars", "id"]
```

Comment:
why is there two batch keys here?

Reply:
There are two because there is a mistake, my bad :cry: . Only the last one is applied `batchKey: ["bars", "id"]`
The `batchKey` is used to group items.

If you have
```json
{
  "bars": [
    {
      "id": 1,
      "bar": "bar 1"
    },
    {
      "id" 2,
      "bar": "bar 2"
    },
    {
      "id" 2,
      "bar": "bar 2_2"
    }
  ]
}
```

The `batchKey` dictates the path that Tailcall will follow to group together items. So, the response is transformed
```json
{
  "1": [
     {
      "id": 1,
      "bar": "bar 1"
    }
  ],
  "2": [
    {
      "id" 2,
      "bar": "bar 2"
    },
    {
      "id" 2,
      "bar": "bar 2_2"
    }
  ]
}
```